packages/core/src/rules/oas2/__tests__/operation-4xx-problem-details-rfc7807.test.ts

```diff
@@ -0,0 +1,111 @@
+import { outdent } from 'outdent';
+import { lintDocument } from '../../../lint';
+import { parseYamlToDocument, replaceSourceWithRef, makeConfig } from '../../../../__tests__/utils';
+import { BaseResolver } from '../../../resolve';
+
+describe('Oas2 operation-4xx-problem-details-rfc7807', () => {
```

Comment:
I'm not completely sure we have to introduce this check for **OAS 2** at all. The rule itself is about the `application/problem+json` content type's shape which was introduced in **OAS 3** I believe. 

@adamaltman ,  @RomanHotsiy , what's your opinion on this?

Reply:
OAS2 had `produces`/`consumes` fields. If this is hard to apply, let's implement it for oas3 only.